test/quoteSpec.js

```
@@ -0,0 +1,240 @@
+        connectorAccount: 'test.example.connie',
+        sourceExpiryDuration: '0'
+      }
+
+      this.plugin.sendMessage = (msg) => {
+        this.response.data.id = msg.data.id
+        this.plugin.emit('incoming_message', this.response)
+        return Promise.resolve(null)
+      }
+    })
+
+    it('should return a quote from the connector', function * () {
+      this.result.expiresAt = moment().format()
+      const response = yield ILQP.quote(this.plugin, this.params)
+      assert.deepEqual(
+        response,
+        this.result)
+    })
+
+    it('should default to getInfo\'s connectors', function * () {
+      // remove manually provided connectors
+      delete this.params.connectors
+
+      this.result.expiresAt = moment().format()
+      const response = yield ILQP.quote(this.plugin, this.params)
+      assert.deepEqual(
+        response,
+        this.result)
+    })
+
+    it('should quote by destination amount', function * () {
+      this.params.destinationAmount = this.params.sourceAmount
+      delete this.params.sourceAmount
+
+      this.result.expiresAt = moment().format()
+      const response = yield ILQP.quote(this.plugin, this.params)
+      assert.deepEqual(
+        response,
+        this.result)
+    })
+
+    it('should return a local quote if destination is local', function * () {
+      this.params.destinationAddress = 'test.example.bob'
+      const response = yield ILQP.quote(this.plugin, this.params)
+
+      delete this.result.connectorAccount // local quote has no connector
+      assert.deepEqual(response,
+        this.result)
+    })
+
+    it('should throw if source and dest amounts are defined', function * () {
+      this.params.destinationAmount = this.params.sourceAmount = '1'
+
+      yield expect(ILQP.quote(this.plugin, this.params))
+        .to.be.rejectedWith(/provide source or destination amount but not both/)
+    })
+
+    it('should return undefined if there are no connectors', function * () {
```

It should throw an error instead. Returning undefined is pretty unhelpful